### DISCUSSION TITLE
Releasing apps

### DISCUSSION THREAD

**Discussion by Ruttie2006** [unoplatform/uno#10125] [Q&A]
I have a WinUI3 Uno project, but if I set the configuration to `Release|x64`, I do get an exe, but running it doesn't do anything.
Can anyone explain to me why?
Did I miss some docs somewhere?

PS: I also tried running the dll with `dotnet ProjectName.Windows.dll`, but I got an `BadImageFormatException`.

**Comment by jeromelaban** [comment]
You can find documentation about WinUI 3/WinAppSDK [here](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/deploy-packaged-apps). By default, Uno apps use `.msix` packaging, you'll need to install the resulting `.msix` file found when "publishing" the app.

**Reply by Ruttie2006** [reply]
When I do that and try to run it, it immediately crashes.

Looking at a (full) .dmp file, I can see an exception:
![image](https://user-images.githubusercontent.com/80394237/195343549-f58b1e9d-ab8f-4793-9c9c-5820f9f92335.png)
Stack says:
```
00000286`c08614e0 00007ffe`2e205941 combase!RoOriginateError+0x51
00000286`c08614e8 00007ffd`18a81ff2 Microsoft_ui_xaml!DirectUI::FrameworkApplication::InvokeOnLaunchActivated+0x19fdf2
00000286`c08614f0 00007ffd`188e3577 Microsoft_ui_xaml!DirectUI::FrameworkApplication::StartOnCurrentThreadImpl+0x9f
00000286`c08614f8 00007ffd`188e3636 Microsoft_ui_xaml!DirectUI::FrameworkApplicationGenerated::StartOnCurrentThread+0x26
00000286`c0861500 00007ffd`18980a4e Microsoft_ui_xaml!DirectUI::WindowsXamlManager::XamlCore::Initialize+0x7e
00000286`c0861508 00007ffd`189808fa Microsoft_ui_xaml!DirectUI::WindowsXamlManager::Initialize+0xaa
00000286`c0861510 00007ffd`18871aa3 Microsoft_ui_xaml!ctl::ComObjectBase::CreateInstanceBase+0x13
00000286`c0861518 00007ffd`18987a0f Microsoft_ui_xaml!ctl::make<DirectUI::WindowsXamlManager>+0x4f
00000286`c0861520 00007ffd`18987959 Microsoft_ui_xaml!DirectUI::WindowsXamlManagerFactory::InitializeForCurrentThreadImpl+0x21
00000286`c0861528 00007ffd`18987917 Microsoft_ui_xaml!DirectUI::WindowsXamlManagerFactory::InitializeForCurrentThread+0x27
00000286`c0861530 00007ffd`189224e6 Microsoft_ui_xaml!DirectUI::FrameworkApplication::StartDesktop+0x16a
00000286`c0861538 00007ffd`189225e3 Microsoft_ui_xaml!DirectUI::FrameworkApplicationFactory::Start+0x63
```
Looking even further, I know it crashes when it reaches
```
            global::Microsoft.UI.Xaml.Application.Start((p) => {
                var context = new global::Microsoft.UI.Dispatching.DispatcherQueueSynchronizationContext(global::Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread());
                global::System.Threading.SynchronizationContext.SetSynchronizationContext(context);
                new App();
            });
```
In `App.g.i.cs` (`Progam.Main`)

**Reply by Ruttie2006** [reply]
Note that, for some reason, running in via VS works fine.

**Reply by jeromelaban** [reply]
This would likely be a question to ask on the WinUI/WinAppSDK repository, but in the meantime, does a "vanilla" application from the WinAppSDK template work properly on your machine?

**Reply by Ruttie2006** [reply]
It seems like it doesn't...
